xds/internal/resolver/xds_resolver_test.go

```diff
@@ -104,12 +107,12 @@ type testClientConn struct {
 }
 
 func (t *testClientConn) UpdateState(s resolver.State) error {
-	t.stateCh.Send(s)
+	t.stateCh.Replace(s)
 	return nil
 }
 
 func (t *testClientConn) ReportError(err error) {
-	t.errorCh.Send(err)
+	t.errorCh.Replace(err)
```

Comment:
Should this use the same channel as above?  The signals are supposed to be mutually exclusive, and at some point we may want to refactor things to make the two conditions happen via the same call.

Reply:
I switched to `Replace()` mainly to workaround go-control-plane's behavior when the xDS client NACKs an update. The go-control-plane keeps resending the same update until it is ACKed. In one of our test scenarios where we test that a NACKed update results in the error being propagated to the channel, the test logic only reads the error out of the `testClientConn` once. But since the go-control-plane keeps sending the same update again and again, the resolver receives the same error from the xDS client and has to push it up to the ClientConn. Without a `Replace()` here, the test will fail with a leaked goroutine.

Changing this to use the same channel will make things more complicated, because I cannot reliably call `Replace()` because I won't know whether the item that I'm pulling from the channel is an *update* or an *error*.

Reply:
So what will we do if we refactor the resolver API to merge ReportError and Update?  We can still use a different channel if the Update reflects an error, I suppose?

Reply:
Yes, that seems doable to me. I think we have this testClientConn implementation in a few other places as well. So, we would have to get to them as well.